api/include/opentelemetry/trace/propagation/HttpTraceContext.cc

```diff
@@ -0,0 +1,129 @@
+            */
+        }
+
+        static SpanContext extractImpl(Getter getter, T &carrier) {
+            nostd::string_view traceParent = getter(carrier, TRACE_PARENT);
+            if (traceParent == NULL) {
+              return SpanContext.getInvalid(); // I need support of this method from SpanContext definition which is currently unavailable it seems
+            }
+
+            SpanContext contextFromParentHeader = extractContextFromTraceParent(traceParent);
+            if (!contextFromParentHeader.isValid()) {
+                // Again, I still need the support of this method from context.
+                return contextFromParentHeader;
+            }
+
+            nostd::string_view traceStateHeader = getter(carrier, TRACE_STATE);
+            if (traceStateHeader == NULL || traceStateHeader.isEmpty()) {
+                return contextFromParentHeader;
+            }
+
+            try {
+                // Again, the trace state problem, should I treat it like a string or dictionary or an encapsuled class? This is not defined in current files
+                TraceState traceState = extractTraceState(traceStateHeader);
+                // Need getter support from SpanContext
+                return SpanContext.createFromRemoteParent(
+                    contextFromParentHeader.getTraceId(),
+                    contextFromParentHeader.getSpanId(),
+                    contextFromParentHeader.getTraceFlags(),
+                    traceState);
+            } catch (IllegalArgumentException e) {
+              std::cout<<"Unparseable tracestate header. Returning span context without state."<<std::endl;
+              return contextFromParentHeader;
+            }
+        }
+    public:
+        List<String> fields() {
+            return FIELDS;
+        }
+
+        void inject(Setter setter, T &carrier, const Context &context) {
+            checkNotNull(context, "context");
+            checkNotNull(setter, "setter");
+
+            Span span = GetCurrentSpan(context);
+            if (span == null || !span.getContext().isValid()) {
+                // We don't have span.getContext() in span.h, should we just use span? As well as acquiring validity. (I do know how to implement them though)
+                return;
+            }
+            injectImpl(span.getContext(), carrier, setter); // span.getContext() function needed
+        }
+
+        Context extract(Getter get_from_carrier, const T &carrier, Context &context) {
+            checkNotNull(context, "context");
+            checkNotNull(carrier, "carrier");
+            checkNotNull(getter, "getter");
+
+            SpanContext spanContext = extractImpl(carrier, getter);
+            return SetSpanInContext(trace.DefaultSpan(spanContext), context); // I actually need a default span class (implemented) here, I don't know who to ask for though. But both python and java version have default span classes though.
+        }
+}
+}
+}
+}
```

Comment:
Please end namespace with a comment, like

```suggestion
}  // namespace trace
```